ui/app/src/components/root/RootLayout.tsx

```diff
@@ -38,6 +38,20 @@ export const RootLayout: React.FunctionComponent<RootLayoutProps> = (props: Root
     getWorkspaces();
   }, [apiCall]);
 
+  const updateWorkspace = (w: Workspace) => {
+    let i = workspaces.findIndex((f: Workspace) => f.id === w.id);
+    let ws = [...workspaces]
+    ws.splice(i, 1, w);
+    setWorkspaces(ws);
+  }
+
+  const removeWorkspace = (w: Workspace) => {
+    let i = workspaces.findIndex((f: Workspace) => f.id === w.id);
+    let ws = [...workspaces];
+    ws.splice(i, 1);
+    setWorkspaces(ws);
+  }
+
   switch (loadingState) {
 
     case 'ok':
```

Comment:
I've added a loadingStates.ts model so we can re-use the enums across the app (see the CreateUpdateResource stuff), might be good to add in here now (although not really part of your PR's concerns so don't worry if not!)